src/Pelagos/Util/MaintenanceMode.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Pelagos\Util;
+
+/**
+ * A utility to check for maintenance mode, and create maintenance mode file.
+ */
+class MaintenanceMode
+{
+    /**
+     * The file name and path where the maintenance file is located.
+     *
+     * @var string
+     */
+    private $fileName;
+
+    /**
+     * Constructor.
+     *
+     * @param string $fileName The filename of the maintenance file.
+     */
+    public function __construct(string $fileName)
+    {
+        $this->fileName = $fileName;
+    }
+
+    /**
+     * Is the system in maintenance mode.
+     *
+     * @return boolean If in maintenance mode.
+     */
+    public function isMaintenanceMode() : bool
+    {
+        return file_exists($this->fileName);
```

Comment:
Maybe we should also require that the file exists and has a value set for either the maintenance_text or maintenance_color before returning true. This would allow us to put an ini file in place and comment out lines, perhaps having commented-out parameters for different situations with pre-agreed-upon wording.  We could also do this renaming the .ini file to something else though. I'm good with it as is, but wanted to see what y'all thought.

Reply:
While this is a ini file, it's really a system file, controller by the system. Being able to put a ini file place should be considered a hack. As a future improvement this library can be used to create this file, and we should set this up in a controller, or command.